go-client/client/app_stub.go

```diff
@@ -0,0 +1,88 @@
+package client
+
+import (
+	"context"
+	"fmt"
+
+	apppb2 "github.com/deephaven/deephaven-core/go-client/internal/proto/application"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type appStub struct {
+	client *Client
+
+	stub apppb2.ApplicationServiceClient
+
+	cancelFunc context.CancelFunc
+}
+
+func newAppStub(client *Client) appStub {
+	return appStub{client: client, stub: apppb2.NewApplicationServiceClient(client.grpcChannel)}
+}
+
+type changeHandler func(update *apppb2.FieldsChangeUpdate)
+
+// Note that the provided context is saved and used to continue reading the stream
+func (as *appStub) listFields(ctx context.Context, handler changeHandler) error {
+	as.cancelListLoop()
```

Comment:
This mirrors the behavior of the python implementation, since there are valid reasons to call FetchTables a second time while an existing loop is running (the first response from FetchTables is _not_ asynchronous and returns immediately, which can be useful). In either case, a lock is acquired to protect the appStub, so this should not significantly affect concurrency.